### PR TITLE
Improve test coverage for pydantic.type_adapter

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: pydantic/type_adapter.py [hunk 1]
```diff
@@ -338,8 +338,7 @@ def json_schemas(
         key_map, definitions = schema_generator_instance.generate_definitions(inputs)
 
         json_schema: dict[str, Any] = {}
-        if definitions:
-            json_schema['$defs'] = definitions
+        json_schema['$defs'] = definitions
```
Comment (hramezani) [inline pydantic/type_adapter.py:341]: Seems the `if definitions` condition is always `True`. So, I did this change to make the coverage 100%

Comment (Kludex) [reply pydantic/type_adapter.py:360]: You need to readd it... 👀 
```
from pydantic.type_adapter import TypeAdapter

_, top_level_schema = TypeAdapter.json_schemas(
    [(int, 'validation', TypeAdapter(int))],
    ref_template='#/components/schemas/{model}',
)
print(top_level_schema)
# {}
```

Comment (hramezani) [reply pydantic/type_adapter.py]: Yeah, you are right.
I've made a test from your example 😃

File: tests/test_json_schema.py [hunk 2]
```diff
@@ -4176,3 +4176,29 @@ def test_core_metadata_get_json_schema():
...
+    assert json_schema['description'] == 'test description'
+
+
+def test_type_adapter_json_schemas_without_definitions():
+    _, top_level_schema = TypeAdapter.json_schemas(
+        [(int, 'validation', TypeAdapter(int))],
+        ref_template='#/components/schemas/{model}',
+    )
+
+    assert 'definitions' not in top_level_schema
```
Comment (Kludex) [inline tests/test_json_schema.py:4204]: Feel free to ignore me hahaha
```suggestion
    _, json_schema = TypeAdapter.json_schemas(
        [(int, 'validation', TypeAdapter(int))],
        ref_template='#/components/schemas/{model}',
    )

    assert 'definitions' not in json_schema
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**